cmd/hub/mcscontroller/controller.go

```diff
@@ -118,15 +121,205 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
+			Type:  es.Service.Type,
+			Ports: es.Service.Ports,
+		},
+	}
+	if apierrors.IsNotFound(err) {
+		err = r.Client.Create(ctx, &expected)
+		if err != nil {
+			log.Error(err, "unable to create the needed Service Import")
+			return err
+		}
+	} else if err == nil {
+		err = r.updateSi(ctx, &si, &expected, log)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Error(err, "Failed to get the needed service import")
+		return err
+	}
+	return nil
+}
+
+func (r *ServiceExportReconciler) updateSi(ctx context.Context, si *mcsv1a1.ServiceImport, expected *mcsv1a1.ServiceImport, log logr.Logger) error {
+	//not sure it's really needed to check name and namespace - I got the si from the Get according to those fields
+	//however its a general func, maybe should be "safe" for other usages(?)
+	if si.Name == expected.Name && si.Namespace == expected.Namespace {
+		si.Spec.Type = expected.Spec.Type
+		//maybe add a function that change the ports more carefully, without overwriting
+		si.Spec.Ports = expected.Spec.Ports
+		err := r.Client.Update(ctx, si)
+		if err != nil {
+			log.Error(err, "unable to update the needed Service Import")
+			return err
+		}
+	}
+	return nil
+}
+
+// delete the corresponding serviceImport of the given serviceExport if exists.
+func (r *ServiceExportReconciler) deleteImport(ctx context.Context, se *mcsv1a1.ServiceExport) error {
+	es := &mcs.ExportSpec{}
+
+	err := es.UnmarshalObjectMeta(&se.ObjectMeta)
+	if err != nil {
+		return err
+	}
+	namespacedName := types.NamespacedName{
+		Namespace: es.Name,
+		Name:      es.Namespace,
+	}
+	log := r.Log.WithValues("service", namespacedName.Name)
+
+	si := mcsv1a1.ServiceImport{}
+	err = r.Client.Get(ctx, namespacedName, &si)
+	si = mcsv1a1.ServiceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: es.Namespace,
+			Name:      es.Name,
+		},
+		Spec: mcsv1a1.ServiceImportSpec{
+			Type:  es.Service.Type,
+			Ports: es.Service.Ports,
+		},
+	}
+	if apierrors.IsNotFound(err) {
+		//the serviceImport dosen't exists - don't do anything
+		return nil
+	} else if err == nil {
+		//delete the service Import
+		r.Client.Delete(ctx, &si)
+	} else {
+		log.Error(err, "Failed to get the service Import to delete")
+		return err
+	}
+	return nil
+}
+
+//update the condition field under serviceExport status
+//based on the already existing method in the agent - might move to lhutil
+func (r *ServiceExportReconciler) updateServiceExportConditions(ctx context.Context, se *mcsv1a1.ServiceExport,
+	conditionType mcsv1a1.ServiceExportConditionType, status corev1.ConditionStatus, reason string, msg string) error {
+	log := r.Log.WithValues("name", se.Namespace+"/"+se.Name)
+	now := metav1.Now()
+	exportCondition := mcsv1a1.ServiceExportCondition{
+		Type:               conditionType,
+		Status:             status,
+		LastTransitionTime: &now,
+		Reason:             (*string)(&reason),
+		Message:            &msg,
+	}
+
+	numCond := len(se.Status.Conditions)
```

Comment:
it is not necessarily the last or one before last?
You should add the condition then trim the array first element if it exceeds the max. 
We should probably be smarter and not remove critical conditions like valid or conflict unless the added condition is the same type

Reply:
So what do you want logic to be? something like this(?):
1. add the condition
2. if exceed max - trim the last "same category" condition (if we added valid than last valid if we added conflict than the last conflict)
3. if we couldn't find it, trim the first one.

Reply:
trim the *oldest* of same kind - not sure what you meant by "last"